### ISSUE TITLE
Implement the Clone trait for Data and HashBuilder

### ISSUE THREAD

**Issue by vktec** [rustache/rustache#148]
Passing `HashBuilder`s around is currently very difficult as `HashBuilder` and `Data` do not implement `Clone`. If this could be implemented, `HashBuilder`s could be defined once and then passed to a function multiple times.

If this is already possible in a different way, please let me know.

**Comment by cardoe** [comment]
That should be possible to do. Would it be better to use a built in collection from `std::collections` instead of `HashBuilder`?

**Comment by vktec** [comment]
Yes, perhaps implementing `Data::from_string`, `Data::from_int`, etc. and using a `std::collections::HashMap` or `std::collections::BTreeMap` (both implement `Clone` and have almost identical APIs) would be a better idea.

I'm not 100% sure why `HashBuilder` exists in the first place...

**Comment by cardoe** [comment]
So looking into this a bit more its because of lambadas being part of the mustache spec. However the change in #150 seems to add the froms you mentioned. Likely we'd just need to add clone but I'm curious how you're passing it around?

**Comment by cardoe** [comment]
Please look at the proposed API in https://github.com/cardoe/rustache/blob/9748a678fe8815d8e9373a5085d53846d06b5326/README.md and let me know if that would make things easier for you?

**Comment by cardoe** [comment]
I'll also note that its not possible to `#[derive(Clone)]` due to the `Lambda` inside of `Data`. I'll certainly merge in any ideas for improvements if you've got a PR.
